[0.040s] Invoking command in '/home/song/steadylab/src/build/serial_communication': CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/home/song/steadylab/install/yo:/home/song/steadylab/install/steadylab:/opt/ros/foxy CONDA_PROMPT_MODIFIER=(zed) /usr/bin/cmake --build /home/song/steadylab/src/build/serial_communication -- -j20 -l20
[0.352s] [ 10%] Built target serial_communication__rosidl_generator_c
[0.352s] [ 12%] Built target serial_communication__cpp
[0.353s] [ 23%] Built target serial_communication__rosidl_typesupport_introspection_c
[0.353s] [ 33%] Built target serial_communication__rosidl_typesupport_c
[0.353s] [ 43%] Built target serial_communication__rosidl_typesupport_cpp
[0.354s] [ 53%] Built target serial_communication__rosidl_typesupport_fastrtps_cpp
[0.354s] [ 64%] Built target serial_communication__rosidl_typesupport_introspection_cpp
[0.356s] [ 74%] Built target serial_communication__rosidl_typesupport_fastrtps_c
[0.356s] [ 74%] Built target serial_communication
[0.357s] [ 76%] [34m[1mGenerating Python code for ROS interfaces[0m
[0.357s] ImportError in template '/opt/ros/foxy/share/rosidl_generator_py/cmake/../resource/_msg.py.em': 
[0.357s] 
[0.357s] IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
[0.358s] 
[0.358s] Importing the numpy C-extensions failed. This error can happen for
[0.359s] many reasons, often due to issues with your setup or how NumPy was
[0.359s] installed.
[0.360s] 
[0.361s] We have compiled some common reasons and troubleshooting tips at:
[0.361s] 
[0.362s]     https://numpy.org/devdocs/user/troubleshooting-importerror.html
[0.363s] 
[0.363s] Please note and check the following:
[0.363s] 
[0.364s]   * The Python version is: Python3.9 from "/home/song/anaconda3/bin/python3"
[0.364s]   * The NumPy version is: "1.23.1"
[0.364s] 
[0.365s] and make sure that they are the versions you expect.
[0.366s] Please carefully study the documentation linked above for further help.
[0.366s] 
[0.366s] Original error was: No module named 'numpy.core._multiarray_umath'
[0.367s] 
[0.367s] ImportError when expanding '_idl.py.em' into '/home/song/steadylab/src/build/serial_communication/rosidl_generator_py/serial_communication/msg/_write_car.py': 
[0.370s] 
[0.370s] IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
[0.371s] 
[0.372s] Importing the numpy C-extensions failed. This error can happen for
[0.372s] many reasons, often due to issues with your setup or how NumPy was
[0.372s] installed.
[0.372s] 
[0.372s] We have compiled some common reasons and troubleshooting tips at:
[0.372s] 
[0.373s]     https://numpy.org/devdocs/user/troubleshooting-importerror.html
[0.373s] 
[0.373s] Please note and check the following:
[0.373s] 
[0.373s]   * The Python version is: Python3.9 from "/home/song/anaconda3/bin/python3"
[0.374s]   * The NumPy version is: "1.23.1"
[0.374s] 
[0.374s] and make sure that they are the versions you expect.
[0.374s] Please carefully study the documentation linked above for further help.
[0.374s] 
[0.374s] Original error was: No module named 'numpy.core._multiarray_umath'
[0.374s] 
[0.375s] Error processing idl file: /home/song/steadylab/src/build/serial_communication/rosidl_adapter/serial_communication/msg/WriteCar.idl
[0.375s] Traceback (most recent call last):
[0.375s]   File "/opt/ros/foxy/lib/python3.8/site-packages/numpy/core/__init__.py", line 23, in <module>
[0.375s]     from . import multiarray
[0.375s]   File "/opt/ros/foxy/lib/python3.8/site-packages/numpy/core/multiarray.py", line 10, in <module>
[0.375s]     from . import overrides
[0.375s]   File "/opt/ros/foxy/lib/python3.8/site-packages/numpy/core/overrides.py", line 6, in <module>
[0.375s]     from numpy.core._multiarray_umath import (
[0.375s] ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'
[0.375s] 
[0.375s] During handling of the above exception, another exception occurred:
[0.376s] 
[0.377s] Traceback (most recent call last):
[0.378s]   File "/opt/ros/foxy/share/rosidl_generator_py/cmake/../../../lib/rosidl_generator_py/rosidl_generator_py", line 40, in <module>
[0.378s]     sys.exit(main())
[0.379s]   File "/opt/ros/foxy/share/rosidl_generator_py/cmake/../../../lib/rosidl_generator_py/rosidl_generator_py", line 36, in main
[0.379s]     return generate_py(args.generator_arguments_file, args.typesupport_impls.split(';'))
[0.379s]   File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_generator_py/generate_py_impl.py", line 61, in generate_py
[0.381s]     generate_files(generator_arguments_file, mapping)
[0.381s]   File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_cmake/__init__.py", line 96, in generate_files
[0.381s]     raise(e)
[0.381s]   File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_cmake/__init__.py", line 87, in generate_files
[0.381s]     expand_template(
[0.382s]   File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_cmake/__init__.py", line 149, in expand_template
[0.382s]     interpreter.string(template_content, template_path, locals=data)
[0.383s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2391, in string
[0.384s]     self.safe(scanner, True, locals)
[0.384s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2401, in safe
[0.384s]     self.parse(scanner, locals)
[0.384s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2421, in parse
[0.384s]     token.run(self, locals)
[0.384s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 1547, in run
[0.385s]     self.subrun(info[0][1], interpreter, locals)
[0.385s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 1631, in subrun
[0.385s]     token.run(interpreter, locals)
[0.385s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 1425, in run
[0.385s]     interpreter.execute(self.code, locals)
[0.385s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2596, in execute
[0.385s]     _exec(statements, self.globals, locals)
[0.385s]   File "<string>", line 2, in <module>
[0.385s]   File "/opt/ros/foxy/lib/python3.8/site-packages/rosidl_cmake/__init__.py", line 198, in _expand_template
[0.385s]     interpreter.string(content, str(template_path), kwargs)
[0.385s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2391, in string
[0.385s]     self.safe(scanner, True, locals)
[0.385s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2401, in safe
[0.385s]     self.parse(scanner, locals)
[0.386s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2421, in parse
[0.386s]     token.run(self, locals)
[0.386s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 1425, in run
[0.386s]     interpreter.execute(self.code, locals)
[0.386s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2596, in execute
[0.386s]     _exec(statements, self.globals, locals)
[0.386s]   File "<string>", line 3, in <module>
[0.386s]   File "/opt/ros/foxy/lib/python3.8/site-packages/numpy/__init__.py", line 140, in <module>
[0.386s]     from . import core
[0.386s]   File "/opt/ros/foxy/lib/python3.8/site-packages/numpy/core/__init__.py", line 49, in <module>
[0.386s]     raise ImportError(msg)
[0.387s] ImportError: 
[0.387s] 
[0.387s] IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!
[0.387s] 
[0.387s] Importing the numpy C-extensions failed. This error can happen for
[0.387s] many reasons, often due to issues with your setup or how NumPy was
[0.387s] installed.
[0.387s] 
[0.387s] We have compiled some common reasons and troubleshooting tips at:
[0.387s] 
[0.387s]     https://numpy.org/devdocs/user/troubleshooting-importerror.html
[0.387s] 
[0.387s] Please note and check the following:
[0.387s] 
[0.387s]   * The Python version is: Python3.9 from "/home/song/anaconda3/bin/python3"
[0.388s]   * The NumPy version is: "1.23.1"
[0.388s] 
[0.388s] and make sure that they are the versions you expect.
[0.388s] Please carefully study the documentation linked above for further help.
[0.388s] 
[0.388s] Original error was: No module named 'numpy.core._multiarray_umath'
[0.388s] 
[0.388s] Exception ignored in: <function Interpreter.__del__ at 0x7effa84383a0>
[0.388s] Traceback (most recent call last):
[0.388s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2094, in __del__
[0.388s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2157, in shutdown
[0.388s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2647, in finalize
[0.388s]   File "/home/song/anaconda3/lib/python3.9/site-packages/em.py", line 2200, in push
[0.388s] AttributeError: '_io.TextIOWrapper' object has no attribute 'push'
[0.388s] make[2]: *** [serial_communication__py/CMakeFiles/serial_communication__py.dir/build.make:82: rosidl_generator_py/serial_communication/_serial_communication_s.ep.rosidl_typesupport_fastrtps_c.c] ì˜¤ë¥˜ 1
[0.389s] make[1]: *** [CMakeFiles/Makefile2:553: serial_communication__py/CMakeFiles/serial_communication__py.dir/all] ì˜¤ë¥˜ 2
[0.389s] make: *** [Makefile:141: all] ì˜¤ë¥˜ 2
[0.390s] Invoked command in '/home/song/steadylab/src/build/serial_communication' returned '2': CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/home/song/steadylab/install/yo:/home/song/steadylab/install/steadylab:/opt/ros/foxy CONDA_PROMPT_MODIFIER=(zed) /usr/bin/cmake --build /home/song/steadylab/src/build/serial_communication -- -j20 -l20
